- Created `User` and `Role` entities

Likh-Alex opened the pull request:
- Created `Repository` and `Service` for `User` and `Role`

Likh-Alex committed:
- Created `User` and `Role` entities
- Created `Repository` and `Service` for `User` and `Role`

Likh-Alex committed:
- deleted `thymeleaf` dependency as redundant

vlykhvar commented on src/main/java/com/dev/bankingservice/entity/Role.java:
```diff
@@ -0,0 +1,28 @@
+    private RoleName roleName;
+
+    public enum RoleName {
+        CUSTOMER,
+        MANAGER
```
RoleName should be ADMIN and USER by requirements

Likh-Alex replied:
Thank you, renamed the role to `ADMIN` and `USER`.

Likh-Alex committed:
- renamed roles to `Admin` and `User`

KseniiaMakarova commented on src/main/java/com/dev/bankingservice/repository/RoleRepository.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Id> {
```
```suggestion
public interface RoleRepository extends JpaRepository<Role, Long> {
```

Likh-Alex replied:
Done

KseniiaMakarova commented on src/main/java/com/dev/bankingservice/repository/RoleRepository.java:
```diff
@@ -0,0 +1,12 @@
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Id> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = :name")
+    Role getByName(@Param("name") Role.RoleName name);
```
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)

Likh-Alex replied:
Thank you, done

KseniiaMakarova commented on src/main/java/com/dev/bankingservice/service/UserService.java:
```diff
@@ -0,0 +1,15 @@
+
+    User getByPhoneNumber(String phoneNumber);
+
+    void update(User user);
```
see checklist

Likh-Alex replied:
removed `update` method and rename `create` to `save` to use for both `create` and `update` operations.

KseniiaMakarova commented on src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,40 @@
+
+    @Override
+    public void delete(Long id) {
+        userRepository.delete(userRepository.getOne(id));
```
```suggestion
        userRepository.deleteById(id);
```

Likh-Alex replied:
Done

KseniiaMakarova replied:
probably you forgot to push the change here?

Likh-Alex replied:
Certainly, fixed it now

Likh-Alex committed:
- removed `Query` in the `RoleRepository`
- removed `update` method and renamed `create` to `save` to use for both `create` and `update` operations.

Likh-Alex committed:
- refactored `getByRoleName` method in `RoleRepository`
- renamed `create` to `save` in `RoleService` interface

KseniiaMakarova commented on src/main/java/com/dev/bankingservice/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+
+    @Override
+    public void deleteById(Long id) {
+        userRepository.delete(userRepository.getOne(id));
```
```suggestion
        userRepository.deleteById(id);
```

Likh-Alex replied:
Now it is correct

Likh-Alex committed:
- used `CrudRepository` `deleteById` method in `UserServiceImpl` instead of `delete`
